### PR TITLE
fix: Fix test case for api

### PR DESCRIPTION
**Pull request by sagar-qa007**
## Description
RCA: Toggle button from log tab is not open always. Due to this next element was not visible. 
Fix: Have gone through the test case and found unnecessary steps for verifying successful log. Update the only required code and fixed the test case.

Fixes #`36082`  

## Automation

/ok-to-test tags="@tag.Datasource"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/10714184973>
> Commit: bd850ca5d80f39c544b6953c85cddc43a3e723d0
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=10714184973&attempt=1" target="_blank">Cypress dashboard</a>.
> Tags: `@tag.Datasource`
> Spec:
> <hr>Thu, 05 Sep 2024 05:42:15 UTC

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [x] No

## Summary by CodeRabbit

- **New Features**
	- Enhanced API testing capabilities with updated test cases for various HTTP methods.
	- Introduced new properties for improved API debugging in the UI.
	- Added a method to check the visibility of specific elements in tests.

- **Bug Fixes**
	- Improved the `validateRequest` command to handle multiple API log entries more effectively.

- **Chores**
	- Updated the test specifications to focus on server-side API tests, replacing outdated client-side entries.

### CODE REVIEW

File: app/client/cypress/support/Pages/AggregateHelper.ts [hunk 1]
```diff
@@ -1552,6 +1552,18 @@ export class AggregateHelper {
     ) as Cypress.Chainable<boolean>;
   }
 
+  IsElementVisibleWithEq(selector: ElementType, eq: number = 0) {
```
[the remark this answers is not in the document]
Comment (sagar-qa007) [reply app/client/cypress/support/Pages/AggregateHelper.ts:1562]: Good suggestion. Added the comment.

File: app/client/cypress/e2e/Regression/ServerSide/ApiTests/API_All_Verb_spec.js [hunk 2]
```diff
@@ -127,12 +112,8 @@ describe(
...
       apiPage.ResponseStatusCheck("200 OK");
       cy.ResponseCheck(testdata.responsetext);
-      cy.validateRequest(
-        "Executed successfully",
-        testdata.baseUrl,
-        testdata.methods,
-        testdata.Get,
-      );
+      agHelper.GetNClickByContains(apiLocators.apiResponseTabsList, "Logs");
+      agHelper.AssertContains("Executed successfully from user request");
```
Comment (ApekshaBhosale) [inline app/client/cypress/e2e/Regression/ServerSide/ApiTests/API_All_Verb_spec.js:116]: is this something you can add to messages and use here? @sagar-qa007

**Review by ApekshaBhosale — CHANGES_REQUESTED** [review]

File: app/client/cypress/e2e/Regression/ServerSide/ApiTests/API_All_Verb_spec.js (hunk 2 above)
Comment (sagar-qa007) [reply app/client/cypress/e2e/Regression/ServerSide/ApiTests/API_All_Verb_spec.js:116]: I understand, it is not a notification message and not stored in constant file anywhere in code. 

However as improvement, I have added string into const in a spec file as it is used only here.

File: app/client/cypress/e2e/Regression/ServerSide/ApiTests/API_All_Verb_spec.js [hunk 3]
```diff
@@ -39,12 +41,8 @@ describe(
...
         apiPage.RunAPI();
         apiPage.ResponseStatusCheck("200 OK");
-        cy.validateRequest(
-          "Executed successfully",
-          testdata.baseUrl,
-          testdata.echoMethod,
-          testdata.Put,
-        );
+        agHelper.GetNClickByContains(apiLocators.apiResponseTabsList, "Logs");
+        agHelper.AssertContains(successMsg);
```
[the remark this answers is not in the document]
Comment (sagar-qa007) [reply app/client/cypress/e2e/Regression/ServerSide/ApiTests/API_All_Verb_spec.js:45]: No need of function as it is just validation.

File: app/client/cypress/e2e/Regression/ServerSide/ApiTests/API_All_Verb_spec.js (hunk 2 above)
Comment (sagar-qa007) [reply app/client/cypress/e2e/Regression/ServerSide/ApiTests/API_All_Verb_spec.js]: Updated the code.

**Review by ApekshaBhosale — APPROVED** [review]

**Pull request merged**